include/cuda/std/atomic

```diff
@@ -159,7 +159,11 @@ struct atomic<_Tp*, _Sco>
     _Tp* operator-=(ptrdiff_t __op) noexcept          {return fetch_sub(__op) - __op;}
 };
 
-inline __host__ __device__ void atomic_thread_fence(memory_order __m, thread_scope _Scope = thread_scope_system) {
+inline __host__ __device__ void atomic_thread_fence(memory_order __m, thread_scope
+#ifdef __CUDA_ARCH__
+_Scope
+#endif
+= thread_scope_system) {
```

Comment:
I wonder if it might be cleaner to just cast the `_Scope` parameter to void inside the `#else`, e.g.,
```
#else
   (void) _Scope:
```

Reply:
I've seen casts fail to sink warnings like this properly. Notably in <tuple> tests. But it's likely it won't be needed here.

https://github.com/NVIDIA/libcudacxx/blob/66c034cf53d8ccd75124f85a5b1dbfa45ac61177/.upstream-tests/test/std/utilities/tuple/tuple.tuple/tuple.creation/tuple_cat.pass.cpp#L38

Because of future work for the nvcxx-compatibility branch I would strongly hesitate to use `__CUDA_ARCH__` macro blocks outside of function scopes. They are not compatible with `if target()` constructs.

So either @jrhemstad suggestion or we think of another way to sink it properly.

Reply:
@jrhemstad @wmaxey I've reverted the changes with macro and added @jrhemstad 's suggestion. It seems to sink the warning